### DISCUSSION TITLE
MSB and LSB for Program Changes

### DISCUSSION THREAD

**Discussion by Jacob-Owen** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#99] [Feature Requests & Ideas]
Can we get MSB and LSB for program changes as it is a struggle with my BOSS pedals to repatch program changes when they have a higher LSB than 1

**Comment by simonaglover95** [comment]
MSB and LSB just means you need to send two messages in a row. Is the BOSS unit not responding to two messages sent one after the other?

**Comment by TheGuacamoleXplosion** [comment]
I guess OP refers to Bank Change MSB and LSB. Technically, these are just CC#0 and CC#32. So, you can send those already now from your BRIDGE4/6 controller.

Of course, it would be a cool feature if the BRIDGE 4/6 scrolling function could somehow include Bank Change when it comes to MIDI Program Changes. Like, have a pair of linked scrolling footswitches than can go through all 200 presets of a Strymon Timeline or Möbius; or all 300 presets of a Strymon Big Sky. But this would require either a very well thought out universal system, and/or device specific templates; since some manufacturers use only MSB Bank Change, some only use LSB, and then there’s stuff like: On Strymon Timeline, MSB=0 has PCs 0 to 127, while MSB=1 has only PCs 0 to 71; so, scrolling up from MSB=1/PC=71 should go to MSB=0/PC=0, and down from there vice versa.

Blogged about this here, including mapping of the MIDI Bank/PC commands for the 3 classic Strymon big boxes:
http://theguacamolexplosion.eu/boss-es-5-tips-iii-building-patches.html#mcetoc_1dd56q18f0
